BufferList.js

```diff
@@ -0,0 +1,344 @@
+'use strict'
+
+var symbol = Symbol.for('BufferList')
+
+function BufferList (buf) {
+  if (!(this instanceof BufferList))
+    return new BufferList(buf)
+
+  BufferList._init.call(this, buf)
+}
+
+BufferList._init = function _init (buf) {
```

Comment:
had to resort to this to avoid circular calls due to the two constructors getting tangled, unfortunately